agents/craftassist/craftassist_swarm_worker.py

```diff
@@ -0,0 +1,80 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+"""
+from droidlet.perception.craftassist.swarm_worker_perception import SwarmLowLevelMCPerception
+from agents.craftassist.craftassist_agent import CraftAssistAgent
+from multiprocessing import Process, Queue
+from droidlet.interpreter.craftassist import tasks
+from droidlet.interpreter.task import ControlBlock
+import logging
+import pdb
+
+TASK_MAP = TASK_MAP = {
+        "move": tasks.Move,
+        "undo": tasks.Undo,
+        "build": tasks.Build,
+        "destroy": tasks.Destroy,
+        "spawn": tasks.Spawn,
+        "fill": tasks.Fill,
+        "dig": tasks.Dig,
+        "dance": tasks.Dance,
+        "point": tasks.Point,
+        "dancemove": tasks.DanceMove,
+        "get": tasks.Get,
+        "drop": tasks.Drop,
+        "control": ControlBlock,
+    }
+
+class CraftAssistSwarmWorker(CraftAssistAgent):
+    def __init__(self, opts, idx=0):
+        self.agent_idx = idx
+        super(CraftAssistSwarmWorker, self).__init__(opts)
+
+    def init_event_handlers(self):
+        pass
+
+    def init_perception(self):
+        """Initialize perception modules"""
+        self.perception_modules = {}
+        self.perception_modules["low_level"] = SwarmLowLevelMCPerception(self)
+
+    def init_controller(self):
+        """Initialize all controllers"""
+        pass
+
+    def perceive(self, force=False):
```

Comment:
does each agent have a full memory system?  which things are stored in worker vs swarm master memory?